plugins/catalog-backend-module-github/src/providers/GithubMultiOrgEntityProvider.ts

```diff
@@ -299,20 +299,24 @@ export class GithubMultiOrgEntityProvider implements EntityProvider {
 
     const allUsers = Array.from(allUsersMap.values());
 
-    const { markCommitComplete } = markReadComplete({ allUsers, allTeams });
-
-    await this.connection.applyMutation({
-      type: 'full',
-      entities: [...allUsers, ...allTeams].map(entity => ({
-        locationKey: `github-multi-org-provider:${this.options.id}`,
-        entity: withLocations(
-          `https://${this.options.gitHubConfig.host}`,
-          entity,
-        ),
-      })),
-    });
+    if (allUsers.length > 0 || allTeams.length > 0) {
```

Comment:
I think we should perhaps not just log the error on line 296 but throw as well? Feel like this is the root cause of the issue you described with GitHub being down.

This makes sense as a safety net, I'm not super familiar with this provider but the one thought I had was: is it possible for either of these to be empty and that's valid? Or do we always pull all the data each run? This would also be something to consider for the other providers being changed here

Reply:
Piggybacking off of this, I agree the error should be checked instead. I _think_ this change would stop a very very small case where you emptied out your Github organization completely and Backstage would be unable to overwrite the existing list of entities.